### ISSUE TITLE
Nginx reverse proxy with basic auth

### ISSUE THREAD

**Issue by gordon848** [jellyfin/jellyfin-media-player#150]
First of all: thanks for this awesome player!

I tried to use jellyfin-media-player with basic auth because I want an extra layer of security. I enter my serverurl "httsp://mydomain.com and a window asks for my credentials (basic auth). Everything seems to work but I cant play anything in my library "media error: loading fails". In the admin panel it shows that the file is being played. If one is using a browser everything works but this isnt a option because I want "Direct play" which works very well with jellyfin-media-player but obviously not inside a webrowser.

 Is there a way to fix this or is it not possible withe the player?

**Comment by iwalton3** [comment]
If you want to try testing a local code change, you can mess with this file, which is installed in the extension directory and is not minified: https://github.com/jellyfin/jellyfin-media-player/blob/master/native/mpvVideoPlayer.js

It is the plug-in for the web player that talks to the player backend.

**Comment by mijofa** [comment]
Ah cool, I'll take a look when I've got some time over the weekend, wasn't sure where to start so that helps, thanks.
I wouldn't expect the Javascript engine to have access to the HTTP auth credentials as generally that could allow external websites to exfiltrate some of that data, but I'll take a look.

**Comment by iwalton3** [comment]
Most of the “brains” of the player are in jellyfin-web. The extensions in that directory communicate through an API channel to the MPV player which is *behind* the web view.

**Comment by iwalton3** [comment]
If you are interested in working on this, please re-open. Likely changes to both JMP and jellyfin-web would be needed.

**Comment by mijofa** [comment]
Ah, I kinda forgot to every look back at this one after I worked around it myself.
My **workaround** was to connect the frontend client via VPN and configure the web server to allow unauthenticated access from local & VPN addresses.

I believe this shouldn't require changes in jellyfin-web as it mostly works, just need to somehow get the HTTP auth credentials to MPV. But this doesn't work on Android clients and such anyway (and in fact is even worse) so not being supported here is reasonable.



Although I'd really prefer if Jellyfin server and all could support http auth better so that we could offload some of that security off to separate tools that are better designed for the job, but that's a whole separate thing

**Comment by sam-6174** [comment]
@mijofa You (and others following this issue) might be interested in how I've worked around this issue:

https://github.com/sam-6174/jellytin

tl;dr All apps (Android, etc.) can access Jellyfin over the internet, while still relying on an identity provider (Authentik) to gate access.

**Comment by iwalton3** [comment]
I agree until I can tighten down Jellyfin security it would be nice to be able to add "extra" authentication. I usually just use IP filtering because of client support.

**Comment by mijofa** [comment]
That looks like a far more complex, but similar set up to what I have:

:bust_in_silhouette:/jellyfin-media-player -> Wireguard on physical server ->  Jellyfin in a VM/container

While other users in a web browser would have to use HTTP auth via:

:bust_in_silhouette:/web-browser -> Nginx on physical server requiring http auth ->  Jellyfin in a VM/container


Both users have to go through a login screen at Jellyfin as well, which I think your solution avoids, but that's an annoyance I'm happy to pay for security, since adding more applications to handle authentication for working around requirements in other applications makes security **worse** not better.
